test/decoders/test_ops.py

```diff
@@ -922,6 +923,76 @@ def get_all_frames(asset, sample_rate=None, stop_seconds=None):
 
         torch.testing.assert_close(frames_downsampled_to_8000, frames_8000_native)
 
+    @pytest.mark.parametrize("buffering", (0, 1024))
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_file_like_decoding(self, buffering, device):
+        # Test to ensure that seeks and reads are actually going through the
+        # methods on the IO object.
+        #
+        # Note that we do not check the number of reads in this test past the
+        # initialization step. That is because the number of reads that FFmpeg
+        # issues is dependent on the size of the internal buffer, the amount of
+        # data per frame and the size of the video file. We can't control
+        # the size of the buffer from the Python layer and we don't know the
+        # amount of data per frame. We also can't know the amount of data per
+        # frame from first principles, because it is data-depenent.
```

Comment:
Not sure if that would help, but we could store the number of samples of each frames in the .json files that we check in